src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect4.java

```diff
@@ -0,0 +1,60 @@
+/*
+JavadocParagraph
+violateExecutionOnNonTightHtml = (default)false
+allowNewlineParagraph = (default)true
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocparagraph;
+
+/**
+ * Some summary.
+ *
+ * <p><h1>Testing...</h1>
+ */
+// violation 2 lines above '<p> tag should not precede HTML block-tag '<h1>''
+public class InputJavadocParagraphIncorrect4 {
+    // violation 4 lines below '<p> tag should not precede HTML block-tag '<ul>''
+    /**
+     * Some summary.
+     *
+     *<p>
+     *  <ul>
+     *    <p>
+     *      <li>1</li> // should NOT give violation as there is not empty line before
+     *  </ul>
+     *
+     *
+     * <p><b>testing</b> // ok, inline HTML tag. Not a block-level tag
```

Comment:
`* <p><b>testing</b> ok, 'b' is inline HTML tag, not a block-level tag.`

no "//"

Reply:
fixed